articles/how-to-replicate-postgresql-database/index.md

```diff
@@ -0,0 +1,218 @@
+```
+
+3. Edit pg_haba.cnf with any nano application in ubuntu and add the configuration
+
+`file edit command`
+
+```bash
+nano /etc/postgresql/10/main/pg_hba.conf
+```
+
+**Configuration**
+
+```bash
+host  replication       replication   MasterIP/24   md5
+```
+
+![postgres-login](engineering-education/how-to-replicate-postgresql-database/pg_hba-edit.png)
+
+`MasterIP is the IP address of the Master Server Computer`
+
+4. Open and Edit postgresql.conf and put the following configuration or uncomment if it is commented in the master server
+file edit command
+
+```bash
+nano /etc/postgresql/10/main/postgresql.conf
+```
+
+**Configuration**
+
+```bash
+listen_addresses = 'localhost,MasterIP'
+
+wal_level = replica
+
+wal_keep_segments = 64
+
+max_wal_senders = 10
+```
+
+5. Restart PostgreSQL in Master main server
+
+```bash
+systemctl restart postgresql
+```
+
+### Setting up Replication in Slave Server
+
+1. Login to PostgreSQL RDMS with the command below
+
+```bash
+su - postgres
+```
+
+2. Stop postgresql service from working to enable us to work on it with the command below
+
+```bash
+systemctl stop postgresql
+```
+
+3. Edit pg\_hba.conf file with this command and add configuration
+
+**Edit Command**
+
+```bash
+nano /etc/postgresql/10/main/pg_hba.conf
+```
+
+**Configuration**
+
+```bash
+host  replication       replication   MasterIP/24   md5
+```
+
+4. Open and Edit postgresql.conf<span style="mso-spacerun:yes">&nbsp;</span>in slave server and put the following configuration or uncomment if it is commented
+
+**Edit Command**
+
+```bash
+nano /etc/postgresql/10/main/postgresql.conf
+```
+
+**Configuration**
+
+```bash
+listen_addresses = 'localhost,SlaveIP'
+
+wal_keep_segments = 64
+
+wal_level = replica
+
+hot_standby = on
+
+max_wal_senders = 10
+```
+
+`SlaveIP is the address of the Slave Server`
+
+5. Access PostgreSQL data directory in the Slave server and remove everything
+
+```bash
+cd /var/lib/postgresql/10/main
+```
+
+```bash
+rm -rfv *
+```
+
+6. Copy PostgreSQL Master Server data Directory files to PostgreSQL Slave Server Data directory. Write this command in Slave Server
+
+```bash
+pg_basebackup -h MasterIP -D /var/lib/postgresql/11/main/ -P -U
+
+replication --wal-method=fetch
+```
+
+7. Enter Master Server postgres password and press Enter.
+
+8. Need for recovery.conf file in data Directory to be created and add the following command
+
+**Edit Command**
+
+```bash
+nano /var/lib/postgresql/10/main/recovery.conf
+```
+
+**Configuration**
+
+```bash
+standby_mode          = 'on'
+
+primary_conninfo      = 'host=MasterIP port=5432 user=replication password=YOUR_PASSWORD'
+
+trigger_file = '/tmp/MasterNow'
+```
+
+`YOUR_PASSWORD is the password for replication user in Master server PostgreSQL created`
+
+9. Start slave PostgreSQL since it had been stopped
+
+```bash
+systemctl start postgresql
+```
+
+10. Try creating any database or table in Master PostgreSQL database and observer in Slave PostgreSQL Database
```

Comment:
Please mention how to do this step with the relevant commands because a beginner might not know how to proceed with this step.